src/output-json.c

```diff
@@ -1190,27 +1189,42 @@ OutputInitResult OutputJsonInitCtx(ConfNode *conf)
             }
 
             if (SCConfLogOpenRedis(redis_node, json_ctx->file_ctx) < 0) {
+                SCFree(json_ctx->file_ctx->sensor_name);
                 LogFileFreeCtx(json_ctx->file_ctx);
                 SCFree(json_ctx);
                 SCFree(output_ctx);
                 return result;
             }
         }
 #endif
+#ifdef HAVE_PLUGINS
         else if (json_ctx->json_out == LOGFILE_TYPE_PLUGIN) {
             ConfNode *plugin_conf = ConfNodeLookupChild(conf,
                 json_ctx->plugin->name);
             void *plugin_data = NULL;
-            if (json_ctx->plugin->Open(plugin_conf, &plugin_data) < 0) {
+            const char *append = ConfNodeLookupChildValue(conf, "append");
+            if (append == NULL)
+                append = DEFAULT_LOG_MODE_APPEND;
+
+            SCPluginFileOptions opts = { .threaded = json_ctx->file_ctx->threaded,
+                .log_dir = ConfigGetLogDirectory(),
+                .append = append };
+            if (json_ctx->file_ctx->threaded) {
+                if (!SCLogOpenThreadedFileFp(opts.log_dir, opts.append, json_ctx->file_ctx, 1)) {
+                    SCLogNotice("can't open");
+                }
```

Comment:
The naming suggests we're opening file pointers here. Is that the case?

Reply:
No -- I'll rename the function to be more indicative of what it's doing -- preparing slots for thread opens.